jupyterlab/commands.py

```diff
@@ -1155,9 +1164,16 @@ def _populate_staging(self, name=None, version=None, static_url=None,
             json.dump(data, fid, indent=4)
 
         # copy known-good yarn.lock if missing
-        lock_path = pjoin(staging, 'yarn.lock')
-        if not osp.exists(lock_path):
-            shutil.copy(pjoin(HERE, 'staging', 'yarn.lock'), lock_path)
+        lock_path = pjoin(staging, 'yarn.lock')        
+        lock_template = pjoin(HERE, 'staging', 'yarn.lock')
+        if self.registry != YARN_DEFAULT_REGISTRY:  # Replace on the fly the yarn repository see #3658
```

Comment:
Note, that if the registry has been set to something non-default, and then been set back to default, a `jupyter lab clean && jupyter lab build` command is needed to reset. Ideally this should be documented.